### PR TITLE
59 object editors

### LINKED ISSUE #59: Support aggregation arguments being objects
For aggregations like `histogram`, where some arguments are objects [1] the form needs to show a label corrrespondent to the argument (e.g. `extended_bounds`) and then indented inputs corresponding to that argument fields (e.g. `min` and `max`).

[1] pseudo-code
```
histogram (
    field: string,
    interval: number
    [, min_doc_count: integer]
    [, extended_bounds: {
      min: number,
      max: number
    }]
    [, keyed: boolean]
    [, missing: number]
    [, order: {
      [_count: string = 'asc'|'desc'],
      [_key: string = 'asc'|'desc'],
      [<subaggr_key>: string = 'asc'|'desc'],
      [<subaggr_key>.<key>: string = 'asc'|'desc'],
    }]
)
```

### PR DESCRIPTION
**Pull request by saabi**
Closes #59

### CODE REVIEW

File: src/node_modules/app/components/elementary/ElasticSearchField.svelte [hunk 1]
```diff
@@ -161,92 +147,84 @@
...
-	<div
-		on:mouseover={displayDocs}
-		on:mouseout={hideDocs}
-	>
-		<BooleanRadios
-			id={id}
-			bind:value={fieldValue}
+<div
+on:mouseover={displayDocs}
+on:mouseout={hideDocs}
```
Comment (mindrones) [inline src/node_modules/app/components/elementary/ElasticSearchField.svelte]: needs indentation

File: src/node_modules/app/components/elementary/ExistenceEditor.svelte [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    export let value = null;
+    export let defaultValue = null;
+
+	const dispatch = createEventDispatcher();
+
+    function handleChange (e) {
+        dispatch('change', e.target.checked ? {} : null);
+    }
+
+    $: console.log(value);
```
Comment (mindrones) [inline src/node_modules/app/components/elementary/ExistenceEditor.svelte]: leftover

File: src/node_modules/app/components/elementary/ObjectEditor.svelte [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	import ESField from './ElasticSearchField.svelte';
+	
+	const dispatch = createEventDispatcher();
+
+	export let id = null;
+	export let dataType;
+	export let required = false;
+	export let value = null;
+    export let defaultValue = null;
```
Comment (mindrones) [inline src/node_modules/app/components/elementary/ObjectEditor.svelte]: there's a mix of tabs and spaces, we should use tabs only, here and elsewhere in the PR if the case

File: src/node_modules/app/elasticsearch/config.js [hunk 4]
```diff
@@ -335,8 +335,9 @@ export const defaultValues = {
 		order: {"_count":"desc"} // ES default
 	},
 	date_histogram: {
-		time_zone: 'UTC', // ES default
+		min_doc_count: 1, // Dapsboard default
```
Comment (mindrones) [inline src/node_modules/app/elasticsearch/config.js]: minor, eventually leave these out for the defaultr PR

File: src/node_modules/app/tsservices/index.js [hunk 5]
```diff
@@ -22,12 +22,15 @@ export function getCompletions (
...
-				.trim(),
+			required: details.displayParts.length > 5
+				? details.displayParts[5].text !== '?'
+				: false,
+			displayText: (details.displayParts.length > 5
+				? details.displayParts.slice(7)
+				: details.displayParts)
+			.map(part => part.text)
+			.join('')
+			.trim(),
```
Comment (mindrones) [inline src/node_modules/app/tsservices/index.js]: this needs to be indented

**Review by mindrones — APPROVED** [review]
Great job! 👍 
Good to merge apart from minors that I noted below.

File: src/node_modules/app/components/elementary/ObjectEditor.svelte (hunk 3 above)
Comment (saabi) [reply src/node_modules/app/components/elementary/ObjectEditor.svelte]: I had linting turned off temporarily and this was a new file. I'll lint before I merge.

File: src/node_modules/app/elasticsearch/config.js (hunk 4 above)
Comment (saabi) [reply src/node_modules/app/elasticsearch/config.js]: You want `UTC` out?

File: src/node_modules/app/tsservices/index.js (hunk 5 above)
Comment (saabi) [reply src/node_modules/app/tsservices/index.js]: Linting flags indentation as incorrect.

File: src/node_modules/app/elasticsearch/config.js (hunk 4 above)
Comment (mindrones) [reply src/node_modules/app/elasticsearch/config.js]: no, sorry I meant all defaults to be moved in the upcoming defaults PR for consistency, I should have selected all the relevant lines

Comment (saabi) [reply src/node_modules/app/elasticsearch/config.js]: Ahh got it.

File: src/node_modules/app/tsservices/index.js (hunk 5 above)
Comment (saabi) [reply src/node_modules/app/tsservices/index.js]: I reworked the formatting to make it clearer. Uploading a screen capture for speed:
image

**Pull request merged**